\documentclass[12pt]{article}
 \usepackage[hcentering,bindingoffset=20mm]{geometry}
 \usepackage{placeins}
 \usepackage[numbib]{tocbibind}
 \usepackage{rotating}
\usepackage[square,sort,comma,numbers]{natbib}
 \usepackage{graphicx}
 \usepackage{tabularx}
 \linespread{1.3}
 \usepackage{gensymb}
\usepackage{longtable}
 \usepackage{lscape}
 \usepackage{url}
 \addtolength{\textwidth}{2cm}
 \addtolength{\hoffset}{-1cm}
 
 
 \addtolength{\textheight}{2cm}
 \addtolength{\voffset}{-1cm}
 \setlength{\parindent}{0pt}
 
\title{Redefiition of the evolutionary relationships within the gonyaulacales - a case study for large data set processing. $^{1}$}
\author{xx}
\date{}

\begin{document}
\maketitle
\paragraph{}Anna Liza Kretzschmar$^{2}$\\
Climate Change Cluster (C3), University of Technology Sydney, Ultimo, 2007 NSW, Australia, anna.kretzschmar@uts.edu.au
\paragraph{}Aaron Darling \\
Ithree institute (i3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\paragraph{}Mathieu Fourment \\
Ithree institute (i3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\paragraph{}Shauna Murray\\ 
Climate Change Cluster (C3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\newpage
\section{Abstract}
\newpage

\section{Introduction}
%protists and environmental importance
Dinoflagellates represent an ancient lineage of the eukaryotic branch of life. Elucidating the evolution of the dinoflagellates is key to answering a number of questions. They play a role in several important ecological processes, as members are represented in aquatic environments, where the cover a diverse niches such as symbionts, parasites and some taxa can cause harmful algal blooms.

 The evolutionary relationship of the dinoflagellates is a key factor in determining how many key environmental process came about, why/how their bizaare genetic structure came about and which traits are adaptive and which are involved in speciation. COuld I ta;k any more out of my arse? Probably not. 
Thus, the genetic structure of the dinoflagalates has been elucidated on several occasions, with various methods and degrees of success. --insert studies--
Phylogenetic studies focusing on the dinoflagellates have frequently hit a roadblock on the order of the gonyaulacales. The problems with their analysis have been many fold, particlualry around long branch attraction and low node support. This is due to a number of factors, commonly incomplete lineage sorting and low node support. The low support values appear to be linked to the depth of the dataset, i.e. how many genes were included, as well as the methodology used. The incomplete lineage sorting is usually represented by long branches in the phylogeny, which is caused by sequence similarity between two unrelated taxa. Another factor could be the inclusion of homologs in the alignment.
Several members are prolific neurotoxin producers and have been implicated in seafood intoxication. The neurotoxins enter the food chain as the protists are consumed, commonly along with the substrate they inhabitate, such as macroalgae, which are then passed from one vector to another to higher trophic levels till they reach humans. These single celled algae are the causative agent for many seafood related illnesses, such as ciguatera fish poisoning (CFP), diarrhetic shellfish poisoning (DSP) and paralytic shell fish poisoning (PSP).
Several studies have attempted to do this with ribosomal DNA genes, mitochondrial genes and protein coding genes by concatenation to varying degrees of success. While the evolutionary relationship of most orders within the dinoflagellates can be estimated with reasonable certainty, one order has consistently escaped scrutiny - the gonyaulacales. Analyses for this order have consistently yielded long branch attraction, low confidence values and inconsistent taxon resolution. Yet illuminating the evolution of the gonyaulaceles is key, as neurotoxin production is prevalent in this order. 
\emph{Alexandrium} and xx spp. produce saxitoxin, a trialkyl tetrahydropurine in structure, which is responsible for PSP. Murray et al. (2012) have demonstrated that the gene complex responsible for saxitoxin production derives from horizontal gene transfer (HGT) from cyanobacteria.
A common type of toxin found within the dinoflagellates, ans especially the gonyaulacales, are polyketide toxins. Specifically the ciguatoxins produced by some \emph{Gambierdiscus} spp. are of interest as CFP is a neglected tropical disease with high associated morbidity, which is predicted to increase in prevalence with the advent of climate change.

%intro dinos and gonya within

%weird genetics of dinos and why this has caused inability to place them correcly

%discuss disagreement between authorities like ncbi and algaebase

%previous studies in phylogenetics probelms incl incomplete lineage sorting 

%adanvtage of *beast over other methods
The methodology to elucidate the evolution of the gonyaulacales is crucial. Previous studies have focused on multi gene concatenation. The problem with this approach is that it treats the genes examined as one long gene with congruent rate of evolution. However different genes evolve at different rates, i.e. can have a divergent evolutionary rate *cite Kubatko, Roche*. Especially long branch attraction is prone to overestimate the evolutionary relationship between taxa in a concatenated approach. Here we propose a novel approach to the problem in two ways : 1) by using the multi species coalescent approach in a Bayesian inference framework; and 2) by using single copy genes to run the phylogenetic inference.

\newpage

\section{Materials and methods}


\subsection*{Transcriptomes}
\emph{Coolia malayensis}, \emph{Gambierdiscus carpenteri}, \emph{Gambierdiscus lapillus},, \emph{G. polynesiensis}, \emph{Gambierdiscus} cf. \emph{silvae} \emph{Ostreopsis ovata}, \emph{Ostreopsis rhodesae}, \emph{Ostreopsis siamensis} and \emph{Thecadinium} cf. \emph{kofoidii} were grown in F/10 medium at 25 - 27 degrees celcius, except for \emph{T.} cf. \emph{kofoidii} which was grown at 18 degrees celcius, and cells were harvested via centrifugation at 350 rcf for 10 minutes at late exponential phase. Trizole was directly added to cells, cells were opened by three cycles of freezing in liquid nitrogen and thaw at 95 degrees celcius, then RNA was extracted as per reccomendation for Trizole by the manufacturers. Quality of RNA was screened by Agilent Bioanalyzer and paired-end sequencing was performed on NextSeq 500 High Output by Ramaciotti, UNSW. \emph{G. lapillus} and \emph{G.} cf. \emph{silvae} sequencing libraries were created with 75bp insert size; \emph{G. carpenterii}, \emph{G. polynesiensis} and \emph{T.} cf. \emph{kofoidii} sequencing libraries were created with 150bp inserts.

RNA seq libraries for all remaining transcriptomes were generated by, and downloaded from, the Marine Microbial Eukaryote Transcriptome Sequencing Project \citep{keeling2014marine}.

\subsection*{Pipeline}
The pipeline is packaged in the Nextflow language for streamlined transferal between high powered computing clusters. Modules are written in the Python 2.7 or bash languages. The workflow is separated into pipeline 1 and pipeline 2. 
\subsubsection*{Pipeline 1}
The input to pipeline 1 were the individual RNA sequencing library which is the processed through FastQC for quality assurance, sequences are trimmed with Trimmomatic and assembled with Trinity v2.4.0. Assemblies were then processed with BUSCO2 with the protist specific library.
The RNA libraries with 150bp inserts generated as part of this study were also subjected to Digital Normalization to pool identical transcripts before assembly.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
\subsubsection*{Pipeline 2}
The input to pipeline 2 was the output of BUSCO2, where all complete single copy genes were identified per transcriptome. Any genes that were present in at least 75 \% the transcriptomes were indexed, the corresponding contig extracted from the assemblies, aligned with hmmer3 and unaligned regions trimmed. Species evolutionary inference was based on Bayesian probability with the *BEAT2 model in BEAST2. The analysis was performed under the WAG substitution model \cite{whelan2001general} with a Gamma distribution for four rate categories. A random local clock was employed \cite{drummond2010bayesian}. Posterior distributions of parameters were approximated after 300,000,000 generations of MCMC runs with 4 heated and one cold chain, sampled ever 5,000 generations  with a burn in of 15\%. The inference was run four times to compare convergence parameters, then log and tree files were merged.  The alignments were  --insert *beast parameters*. Inference was run using BEAGLE \cite{ayres2011beagle} on the University of Technology's High-powered computing cluster.

\newpage
\section{Results}

\FloatBarrier
\begin{longtable}{  | p{3cm} |p{4.5cm} | p{2cm} | p{2cm} | p{3cm} |}
\caption{Transcriptomes used for study along with taxonomic placement at family level and source. Family level placement derived from algaebase. MMETSP abbreviation for marine Microbial eukaryotic transcriptome sequencing project, by Moore Foundation.}\\
\hline
\label{tbl:Transcriptomes}
\textbf{Family}&\textbf{Species}&\textbf{Strain}&\textbf{BUSCO single complete}&\textbf{Source}\\
\hline
 \multicolumn{5}{| c |}{Gonyaulacales transcriptomes}\\
    \hline
   Ceratiaceae&\emph{Ceratium fusus}&PA161109&81&MMETSP1074 \citep{keeling2014marine}\\
        \hline
  Crypthecodiniaceae&\emph{Crypthecodinium cohnii}&Seligo&98&MMETSP0326\_2 \citep{keeling2014marine}\\
        \hline
    &\emph{Alexandrium catenella}&OF101&74&MMETSP0790 \citep{keeling2014marine}\\
        \hline
    &\emph{Alexandrium monilatum}&JR08&74&MMETSP0093 \citep{keeling2014marine}\\
        \hline
&\emph{Pyrodinium bahamense}&pbaha01&89&MMETSP0796 \citep{keeling2014marine}\\
        \hline
Gonyaulacaceae&\emph{Coolia malayensis}&MAB&100&This study\\
\hline
&\emph{Gambierdiscus carpenteri}&UTSMER9A&83&This study\\
\hline
&\emph{Gambierdiscus excentricus}&VGO790&83&\\
        \hline
    &\emph{Gambierdiscus lapillus}&HG4&98&This study\\
        \hline
            &\emph{Gambierdiscus polynesiensis}&CG14 really&81&This study\\
        \hline
    &\emph{Gambierdiscus} cf. \emph{silvae}&HG5&87&This study\\
        \hline
    &\emph{Gonyaulax spinifera}&CCMP409&53&MMETSP1439 \citep{keeling2014marine}\\
        \hline
%    &\emph{Lingulodinium polyedra}&CCMP1738&MMETSP1032&209&\citep{keeling2014marine}\\
 %       \hline
     &\emph{Ostreopsis ovata}&HER27&99&This study\\
     \hline
     &\emph{Ostreopsis rhodesae}&HER26&98&This study\\
     \hline
     &\emph{Ostreopsis siamensis}&BH1&98&This study\\
     \hline     
Protoceratiaceae&\emph{Protoceratium reticulatum}&CCCM535=CCMP1889&72&MMETSP0228 \citep{keeling2014marine}\\
    \hline
 &\emph{Thecadinium kofoidii}&THECA&70&This study\\
 \hline
 \multicolumn{5}{| c |}{Outgroup transcriptomes}\\
 \hline
Dinophysiaceae&\emph{Dinophysis acimunata}&DAEP01&74&MMETSP0797 \citep{keeling2014marine}\\
        \hline
Dinophyceae incertae sedis&\emph{Azadinium spinosum}&3D9&81&MMETSP1036\_2 \citep{keeling2014marine}\\
        \hline
Gymnodiniales&\emph{Karenia brevis}&CCMP2229&85&MMETSP0030 \citep{keeling2014marine}\\
%    \hline
%    &\emph{Amphidinium massartii}&&&&\\
%        &\emph{Durinskia baltica}&&&&\\
    \hline
%    &\emph{Peridinium aciculferum}&&&&\\
\end{longtable}

Azadinium Order Dinophyceae incerta sedis according to algaebase, gonyaulacales from NCBI taxonomy

%\FloatBarrier 
%\begin{figure} 
%\includegraphics[scale=.3]{HG7-env.png} 
%\caption{Detection of \emph{G. lapillus} per spatial replicate at each sampling site, cell numbers as normalised to HG7 standard curve (Fig. ~\ref{fig:lapiStd}B). Spatial replicates coloured as per macroalgal substrate,\ where \emph{Chnoospora} sp. are green, \emph{Sargassum} sp. are blue, \emph{Padina} sp. are red and mixed macroalgal substrates are yellow (see table ~\ref{tbl:MacroalgaeTable}).} 
%\label{fig:envHG7}
%\end{figure} 
%\FloatBarrier

\newpage
\section{Discussion}
This study presents a pipeline designed to handle the identification of single copy genes in ordinarily difficult taxa, extracting and aligning single copy gene sequences and running phylogenetic inference. 
Dinoflagellates are renowned for their large genomes and paralogy, which is why historically the evolutionary inference for them has been shithouse. With the recent completion and public availability of the MMETSP database, transcriptomes to investigate the evolutionary relationship for marine protists has become possible. And several studies have looked at the evolutionary relationships. While most clades of the dinoflagellates have been relatively well worked out, the gonyaulacales have been recalcitrant. So these have been of particular interest, especially as many of the taxa produce polyketide toxins that cause seafood intoxications globally. However, now that the transcriptomes are available, it's crucial to develop robust and reproducible methods to analyse the data
\subsection*{Phylogeny}
- differences between ML and BI analyses incl tanglegram
- literature on concatenation problem, incl. BS misleading (Kubatko 07, Roche 15)
- assembly diffs if transrate pans out
- something about single copy genes for paralog riddled taxa

\subsection*{Diff to previous gonyaulacales phylogenies}

The gonyaulacales phylogenies pre-dating this study differ extensively from the results presented here. As established, this is likely due to the methodologies used. A further factor is the available data for evolutionary inference. Before the MMETSP project, studies would either utilise a samll number of taxons with in depth sequencing, or taxons to cover the proposed family structure for this order but utulise only a small number of genes for investigation. The MMETSP database has allowed protists to join many other organisms on the big data set stage. With the access to data no longer as the limiting factor for analyses, the focus must now fall on the methods used to explore this data.
Several studies have been oublished employing the MMETSP database content since the project has gone live, for both direct exploration of the availavble data as well as supplementing trnascriptomes to answer questions. Arguably, there is a dosconnect between bioinformaticians, computer scientists and biologists which is well showcased in this area. The tennets of reproducible bioinformatics is hinged on open access software, as well as documentation, a sound understanding of the methods employed and best practices vs. computational trade-offs in analyses. As the gonyaulacales have proven a recalcitrant order for which to infer evolution, with a number of genetic peculiarities which make nalyses particularly prone to errors, this is an excellent case study.
The phylogeny presented by Price t al. 2017 is based on a similar assumption as this study - that single copy genes are the key to circumventing the issue of paralogy when selecting genes of interest. However, the methodology for identifying and selecting these single copy genes is not included in the publication. Further, contact with the author established that there was no script or documented steps for the pipeline that the authors executed. The bootstrap values for the resulting phylogeny are remarkably well supported, however without transparency for the gene selection for which genes were included or how the sequences were manipulated prior to phylogenetic inference the bootstrap values could simply be a case of the xxx phenomenon described by Kubatko et al. REF.
In Jackounevceck et al.'s 2016 study, the gonyaulacales phylogeny was part of the greater analyses around the phylum ?????, using the MMETSP transcriptomes to infer a number of evolutionary trends such as thecate evolution XXXXXX. NEED TO LOOK THIS HSIT UP AGAIN

The dominant method for established species trees is based on SSU and LSU rDNA sequences, usually within the genus and a small number of outgroups. The prevailing problem with this approach is that that deeper branches bwetween genera or higher taxonomic points is not well explored. Interestingly, the phylogenetic resolution of two genera with the highest species coverage, \emph{Gambierdiscus} and \emph{Ostreopsis}, did not mimic the resolution established in the literature within either genus. This is likely due to the rDNA sequences targeted for the usual studies reflect the evolution of that particular gene region, rather than the species evolution itself. 

- commonly used/required databases for taxon classification by journals, eg algaebase, and how the families don't match phylo
- Hoppenrath 17 suggesting reclassification as family 'asymetricomorpha' based on morphology, discuss differences and similarities
- diff of phylo to species trees within genera, eg. Ostreopsis and Gamb SSU, usually based on rDNA ie. gene like rather than species evolution
%- Saldarriaga 04 \cite{saldarriaga2004molecular} SSU & LSU gene based. ERGH neighbourhood joining method. Table 2: Gonya res as per Fensome 93
- Murray 05 \cite{murray2005improving} rDNA and only some gonya but looks like it meshes
 -orr 12 \cite{orr2012naked} puts Crypthecodinium in the Peridiniales too (Fig4)
 - SHALCHIAN-TABRIZI 06 \cite{shalchian2006combined} alsp places Crythecodinium in the Peridiniales(Fig 3)

Good for discussing findings which are different to literature to date Arroyave '11 Phylogenetic relationships and the temporal context for the diversification
of African characins of the family Alestidae (Ostariophysi: Characiformes): Evidence from DNA sequence data
- Zhang 07 use SSU cox1 and cob but only one or two gonya
- bachvaroff '14 74 protein matrix for phylo. they used concatenation approach and booted duplicate copies by selecting longest and some other rationales. C, chonii wasn't well resolved for them. Only moderate support for gonyaulacpods and prorocentrales but support dropped when gblocks trimming was put into effect. A.tamarenseshowed entirely differently placed copies of gene suggesting gene duplication confounding the analysis.  ref Bachvaroff and Place, 2008; Bachvaroff et al.,2009; Shoguchi et al., 2013 for gene duplication
-bachvaroff '11 concatenation approach 17 rDNA genes. only one gonya.
-derelle '16 stramenopile phylo but use concatenation which behaves badly with recombination/incomplete lineage sorting.
-price '17 talk about method problem
- the big other paper that ignore Shauna's findings





\newpage
\section{Conclusion}
\newpage

\section{Acknowledgements}
\bibliographystyle{acm}
\bibliography{/home/nurgling/PhD/writing/review/references.bib}


\end{document}