\documentclass[12pt]{article}
 \usepackage[hcentering,bindingoffset=20mm]{geometry}
 \usepackage{placeins}
 \usepackage[numbib]{tocbibind}
 \usepackage{rotating}
\usepackage[square,sort,comma,numbers]{natbib}
 \usepackage{graphicx}
 \usepackage{tabularx}
 \linespread{1.3}
 \usepackage{gensymb}
 \usepackage{longtable}
 \usepackage{lscape}
 \usepackage{url}
 \addtolength{\textwidth}{2cm}
 \addtolength{\hoffset}{-1cm}
 
 
 \addtolength{\textheight}{2cm}
 \addtolength{\voffset}{-1cm}
 \setlength{\parindent}{0pt}
 
\title{Gonyalacales phylogeny. $^{1}$}
\author{me}
\date{}

\begin{document}
\maketitle
\paragraph{}Anna Liza Kretzschmar$^{2}$\\
Climate Change Cluster (C3), University of Technology Sydney, Ultimo, 2007 NSW, Australia, anna.kretzschmar@uts.edu.au
\paragraph{}Aaron Darling \\
Ithree institute (i3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\paragraph{}Shauna Murray\\ 
Climate Change Cluster (C3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\newpage
\section{Abstract}
\newpage

\section{Introduction}
Dinoflagellates represent an ancient lineage of the eukaryotic branch of life. Elucidating the evolution of the dinoflagellates is key to answearing a number of questions. They play a role in several important ecological processes, as members are represented in both the marine and sweet water environments, where the cover a diverse nieches such as symbionts, parasites and harmful algal blooms.
%protists and environmental importance
 The evolutionary relationship of the dinoflagellates is a key factor in determining how many key ehnvironemntal process came about, why/how their bizaare genetic structure came about and which traits are adaptive and which are involved in speciation. COuld I ta;k any more out of my arse? Probably not. 
Thus, the genetic structure of the dinoflagalates has been elucidated on several occasions, with various methods and degrees of success. --insert studies--
Phylogenetic studies focusing on the dinoflagellates have frequently hit a roadblock on the order of the gonyaulacales. The problems with their analysis have been many fold, particlualry around long branch attraction and low node support. This is due to a number of factors, commonly incomplete lineage sorting and low node support. The low support values appear to be linked to the depth of the dataset, i.e. how many genes were included, as well as the methodology used. The incomplete lineage sorting is usually represented by long branches in the phylogeny, which is caused by sequence similarity between two unrelated taxa. Another factor could be the inclusion of homologs in the alignment.
Several members are prolific neurotoxin producers and have been implicated in seafood intoxication. The neurotoxins enter the food chain as the protists are consumed, commonly along with the substrate they inhabitate, such as macroalgae, which are then passed from one vector to another to higher trophic levels till they reach humans. These single celled algae are the causative agent for many seafood related illnesses, such as ciguatera fish poisoning (CFP), diarrhetic shellfish poisoning (DSP) and paralytic shell fish poisoning (PSP).
Several studies have attempted to do this with ribosomal DNA genes, mitochondrial genes and protein coding genes by concatenation to varying degrees of success. While the evolutionary relationship of most orders within the dinoflagellates can be estimated with reasonable certainty, one order has consistently escaped scrutiny - the gonyaulacales. Analyses for this order have consistently yielded long branch attraction, low confidence values and inconsistent taxon resolution. Yet illuminating the evolution of the gonyaulaceles is key, as neurotoxin production is prevalent in this order. 
\emph{Alexandrium} and xx spp. produce saxitoxin, a trialkyl tetrahydropurine in structure, which is responsible for PSP. Murray et al. (2012) have demonstrated that the gene complex responsible for saxitoxin production derives from horizontal gene transfer (HGT) from cyanobacteria.
A common type of toxin found within the dinoflagellates, ans especially the gonyaulacales, are polyketide toxins. Specifically the ciguatoxins produced by some \emph{Gambierdiscus} spp. are of interest as CFP is a neglected tropical disease with high associated morbidity, which is predicted to increase in prevalence with the advent of climate change.

%intro dinos and gonya within

%weird genetics of dinos and why this has caused inability to place them correcly

%discuss disagreement between authorities like ncbi and algaebase

%previous studies in phylogenetics probelms incl incomplete lineage sorting 

%adanvtage of *beast over other methods
The methodology to elucidate the evolution of the gonyaulacales is crucial. Previous studies have focused on multi gene concatenation. The problem with this approach is that it treats the genes examined as one long gene with congruent rate of evolution. However different genes evolve at different rates, i.e. can have a divergent evolutionary rate. Especially long branch attraction is prone to overestimate the evolutionary relationship between taxa in a concatenated approach. Here we propose a novel approach to the problem in two ways : 1) by using the multi species coalescent approach in a Bayesian inference framework; and 2) by using single copy genes to run the phylogenetic inference.

\newpage

\section{Materials and methods}

\FloatBarrier
\begin{table}
\caption{Transcriptomes used for study along with taxonomic placement at family level and source. Family level placement derived from algaebase. MMETSP abbreviation for marine Microbial eukaryotic transcriptome sequencing project, by Moore Foundation.}
\label{tbl:Transcriptomes}
\begin{tabular}{  | p{3cm} |p{4cm} | p{2cm} | p{3cm} | p{2cm} | p{2cm} | p{2cm} |}
\hline
\textbf{Family}&\textbf{Species}&\textbf{Strain}&\textbf{ID}&\textbf{CEGMA score}&\textbf{BUSCO single complete}&\textbf{Reference}\\
\hline
 \multicolumn{7}{| c |}{Gonyaulacales transcriptomes}\\
    \hline
   Ceratiaceae&\emph{Ceratium fusus}&PA161109 &MMETSP1074&79.44&202&\citep{keeling2014marine}\\
        \hline
  Crypthecodiniaceae&\emph{Crypthecodinium cohnii}&Seligo&MMETSP0326\_2&83.87&202&\citep{keeling2014marine}\\
        \hline
    &\emph{Alexandrium catenella}&OF101&MMETSP0790&62.10&178&\citep{keeling2014marine}\\
        \hline
    &\emph{Alexandrium monilatum}&JR08&MMETSP0093&77.82&208&\citep{keeling2014marine}\\
        \hline
&\emph{Pyrodinium bahamense}&pbaha01&MMETSP0796&79.84&197&\citep{keeling2014marine}\\
        \hline
Gonyaulacaceae&\emph{Coolia malayensis}&&&&&\\
&\emph{Gambierdiscus carpenteri}&&&&&\\
&\emph{Gambierdiscus excentricus}&&&&&\\
        \hline
    &\emph{Gambierdiscus lapillus}&HG4&This study&85.08&227& \\
        \hline
            &\emph{Gambierdiscus polynesiensis}&CG14&This study&&& \\
        \hline
    &\emph{Gambierdiscus} cf. \emph{silvae}&HG5&This study&77.42&215& \\
        \hline
    &\emph{Gonyaulax spinifera}&CCMP409&MMETSP1439&67.34&131&\citep{keeling2014marine}\\
        \hline
    &\emph{Lingulodinium polyedra}&CCMP1738&MMETSP1032&77.82&209&\citep{keeling2014marine}\\
        \hline
     &\emph{Ostreopsis ovata}&&&&&\\
     \hline
     &\emph{Ostreopsis rhodesae}&&&&&\\
     \hline
     &\emph{Ostreopsis siamensis}&&&&&\\
     \hline     
Protoceratiaceae&\emph{Protoceratium reticulatum}&CCCM535=CCMP1889&MMETSP0228&61.69&176&\citep{keeling2014marine}\\
    \hline
 &\emph{Thecadinium kofoidii}&&&&&\\
 \hline
    \multicolumn{7}{| c |}{Peridiniales transcriptomes}\\
 \hline
 \multicolumn{7}{| c |}{Outgroup transcriptomes}\\
 \hline
Dinophysiaceae&\emph{Dinophysis acimunata}&DAEP01&MMETSP0797&63.71&178&\citep{keeling2014marine}\\
        \hline
Dinophyceae incertae sedis&\emph{Azadinium spinosum}&3D9&MMETSP1036\_2&80.65&193&\citep{keeling2014marine}\\
        \hline
Gymnodiniales&\emph{Karenia brevis}&CCMP2229&MMETSP0030&80.65&184&\citep{keeling2014marine}\\
    \hline
    &\emph{Amphidinium massartii}&&&&&\\
        &\emph{Durinskia baltica}&&&&&\\
    \hline
    &\emph{Peridinium aciculferum}&&&&&\\
\end{tabular}
\end{table}
Azadinium Order Dinophyceae incerta sedis according to algaebase, gonyaulacales from NCBI taxonomy

\FloatBarrier
\begin{table}
\caption{Transcriptomes used for study along with taxonomic placement at family level and source. Family level placement derived from algaebase. MMETSP abbreviation for marine Microbial eukaryotic transcriptome sequencing project, by Moore Foundation.}
\label{tbl:Transcriptomes}
\begin{tabular}{  | p{3cm} |p{4cm} | p{2cm} | p{3cm} | p{3cm}  |}
\hline
\textbf{Family}&\textbf{Species}&\textbf{ID}& \textbf{region} &\textbf{Comment}\\
\hline
Ceratiaceae& Ceratium fusus&AF022153&SSU&\\
\hline
&Ceratium fusus&AF260390&D1-D3&\\
\hline
&Ceratium tripos&AF260389&D1-D3&Maybe Neoceratium or Tripos\\
&Tripos longipes&DQ388462&SSU&ex Ceratium, ex Neoceratium - WORMS\\
\hline
&Neoceratium platycorne&FJ824911&SSU&Fuck knows\\
\hline
Gonyaulacaceae&Amylax triacantha&AB375869&SSU&\\
\hline
&Amylax triacantha&JX666362&D1-D3&\\
\hline
&\emph{Gonyaulax spinifera}&DQ151558&D1-D3 LSU&\\
\hline
&\emph{Gonyaulax spinifera}&AF022155&SSU&\\
\hline
&Lingulodinium polyedrum&AY421788&SSU&\\
\hline
&Lingulodinium polyedrum&EF613357&D1-D3&\\
\hline
Ostreopsidaceae&\emph{Alexandrium catenella}&AB088238&D1-D2&\\
\hline
&\emph{Alexandrium catenella}&AB088286&SSU&\\
\hline
&\emph{Alexandrium monilatum}&AY883005&SSU&\\
\hline
&\emph{Coolia malayensis}&KX589143&D1-D3&\\
\hline
&\emph{Coolia malayensis}&KU746833&SSU&\\
\hline
&\emph{Gambierdiscus carpenteri}&&&\\
\hline
&\emph{Gambierdiscus carpenteri}&&&\\
\hline
&\emph{Gambierdiscus excentricus}&&&\\
\hline
&\emph{Gambierdiscus excentricus}&&&\\
\hline
&\emph{Gambierdiscus lapillus}&&&\\
\hline
&\emph{Gambierdiscus lapillus}&&&\\
\hline
&\emph{Gambierdiscus polynesiensis}&&&\\
\hline
&\emph{Gambierdiscus polynesiensis}&&&\\
\hline
&\emph{Gambierdiscus} cf. \emph{silvae}&&&\\
\hline
&\emph{Gambierdiscus} cf. \emph{silvae}&&&\\
\hline
&\emph{Ostreopsis ovata}&&&\\
\hline
&\emph{Ostreopsis ovata}&&&\\
\hline
&\emph{Ostreopsis rhodesae}&&&\\
\hline
&\emph{Ostreopsis rhodesae}&&&\\
\hline
&\emph{Ostreopsis siamensis}&&&\\
\hline
&\emph{Ostreopsis siamensis}&&&\\
\hline
&\emph{Pyrodinium bahamense}&&&\\
\hline
&\emph{Pyrodinium bahamense}&&&\\
\hline
Protoceratiaceae&Ceratocorys horrida&AF022154&SSU&\\
\hline
&\emph{Protoceratium reticulatum}&&&\\
\hline
&\emph{Protoceratium reticulatum}&&&\\
\hline
Pyrophacaceae&\textit{Fragilidium subglobosum}&AF033869&SSU&\\
\hline
&\textit{Fragilidium subglobosum}&AF260387&D1-D3&\\
\hline
&\textit{Pyrophacus steinii}&EF613363&D1-D3&\\
\hline
&\textit{Pyrophacus steinii}&AY443024&SSU&\\
\hline
Thecadiniaceae&\emph{Thecadinium kofoidii}&&&\\
\hline
&\emph{Thecadinium kofoidii}&&&\\
\hline
Other&\textit{Dapsilidinium pastielsii}&AB919107&&\\
&\textit{Dapsilidinium pastielsii}&AB919106&SSU&\\
&\textit{Heterodinium globosum}&JQ446586&SSU&Heterodiniaceae, fam in Gonyaulacales in WORMS\\
\hline
\end{tabular}
\end{table}

\newpage
\section{Results}

%\FloatBarrier 
%\begin{figure} 
%\includegraphics[scale=.3]{HG7-env.png} 
%\caption{Detection of \emph{G. lapillus} per spatial replicate at each sampling site, cell numbers as normalised to HG7 standard curve (Fig. ~\ref{fig:lapiStd}B). Spatial replicates coloured as per macroalgal substrate,\ where \emph{Chnoospora} sp. are green, \emph{Sargassum} sp. are blue, \emph{Padina} sp. are red and mixed macroalgal substrates are yellow (see table ~\ref{tbl:MacroalgaeTable}).} 
%\label{fig:envHG7}
%\end{figure} 
%\FloatBarrier

\newpage
\section{Discussion}
Good for discussing findings which are different to literature to date Arroyave '11 Phylogenetic relationships and the temporal context for the diversification
of African characins of the family Alestidae (Ostariophysi: Characiformes): Evidence from DNA sequence data
- Zhang 07 use SSU cox1 and cob but only one or two gonya
- bachvaroff '14 74 protein matrix for phylo. they used concatenation approach and booted duplicate copies by selecting longest and some other rationales. C, chonii wasn't well resolved for them. Only moderate support for gonyaulacpods and prorocentrales but support dropped when gblocks trimming was put into effect. A.tamarenseshowed entirely differently placed copies of gene suggesting gene duplication confounding the analysis.  ref Bachvaroff and Place, 2008; Bachvaroff et al.,2009; Shoguchi et al., 2013 for gene duplication
-bachvaroff '11 concatenation approach 17 rDNA genes. only one gonya.
-derelle '16 stramenopile phylo but use concatenation which behaves badly with recombination/incomplete lineage sorting.
-price '17 talk about method problem
- the big other paper that ignore Shauna's findings




\newpage
\section{Conclusion}
\newpage

\section{Acknowledgements}
\bibliographystyle{acm}
\bibliography{gonya.bib}


\end{document}